pennylane/qnn/keras.py

```diff
@@ -327,6 +327,7 @@ def __init__(
 
         # no point in delaying the initialization of weights, since we already know their shapes
         self.build(None)
+        self._initialized = True
```

Comment:
what is this needed for? if you have an instance of something, it's been initialized. Unless there are classmethods that check this, but I don't think you're using any

Reply:
It's needed specifically for `self.__getattr__` and `self.__setattr__`. In the `__init__` method we are getting/setting attributes of the class which don't exist yet, since the class object itself doesn't exist yet. If the object is not initialized yet, we use the default behaviour of `__getattr__` and `__setattr__`.

Reply:
For `__setattr__` calling `self.__dict__[item]` is the default implementation, so we use it as a fallback. For `__getattr__`, I'm not super sure. If I remove it I get some cryptic errors in the base class (`tf.keras.layers.Layer`), so it's most likely something to do  with the base class `__dict__`. 

This is actually the recommended way to implement the `__getattr__` and `__setattr__` methods to prevent anything going wrong in `__init__`.